app2.js

```diff
@@ -60,79 +55,99 @@ newGameBtn.addEventListener("click", () => {
 
-const addScore = (num) => {
-    if(currentPlayer === 1) {
-        player1Score = num + holdScore1.textContent;
-        console.log(player1Score);
-        currentScore1.textContent = player1Score;
-    } else {
-        currentScore2.textContent = num + holdScore2.textContent;
-    }
-    console.log(player1Score);
-    console.log(player2Score);
-}
-    
-const game = (num) => {
-    if(player1Score && player2Score < 20) {
-        addScore(num);
-        console.log(player1Score);
-        console.log(player2Score);
-    } else if (player1Score >= 20) {
-        currentScore1.textContent = "You Win!";
-    } else if (player2Score >= 20) {
-        currentScore2.textContent = "You Win!";
-    }
-    
-}
-
-const rollBtn = document.querySelector("#rollBtn");
+//holdScore un ustune ekleme yapamiyor
 
 rollBtn.addEventListener("click", () => {
     rollDice();
     currentDice = rollDice();
     diceStyling(currentDice);
     console.log(currentDice);
-    game(currentDice);
+    
+    if (currentDice !== 1 && hold1Score === 0) {    
+        if (currentPlayer == 1) {
+            player1Score = player1Score + currentDice;
+            currentScore1.textContent = player1Score;
+        } else {
+            player2Score = player2Score + currentDice;
+            currentScore2.textContent = player2Score;
+        }
+    } else if (currentDice !== 1 && hold1Score !== 0) {    
+        if (currentPlayer == 1) {
+            let afterHoldScore1 = 0;
+            player1Score = hold1Score + currentDice ;
+            currentScore1.textContent = player1Score;
+        } else {
+            player2Score = hold2Score + currentDice;
+            currentScore2.textContent = player2Score;
+        }
+    } else if (currentDice === 1) {
+        currentScore1.textContent = (currentPlayer === 1) ? "You Lost" : "You Win!";
+        currentScore2.textContent = (currentPlayer === 2) ? "You Lost" : "You Win!";
+    }
+    doesWin();
 });
 
-const holdScore1 = document.querySelector("#holdScore1");
-const holdScore2 = document.querySelector("#holdScore2");
+holdBtn.addEventListener("click", () => {
+    holdFunc();
+});
 
-const holdFunc = () => {
-    //ternary operator if/else as below
-    currentPlayer = (currentPlayer === 1) ? 2 : 1;
-    // everything on the left of there is the test
-    // left is the true branch
-    // right side is after colon is False branch
-    switchBg();
 
+//holdScore kendisine eklenmesi gerekiyor
+
+const holdFunc = () => {
     if(currentPlayer === 1){
+        hold1Score = player1Score;
         holdScore1.textContent = player1Score;
         currentScore1.textContent = 0;
-   
+        p1Bg.style.backgroundColor = "transparent";
+        p2Bg.style.backgroundColor = "white";
     } else {
+        hold2Score = player2Score;
         holdScore2.textContent = player2Score;
         currentScore2.textContent = 0;
+        p1Bg.style.backgroundColor = "transparent";
+        p2Bg.style.backgroundColor = "white";
```

Comment:
This is the same as 103 & 104, is this a bug or just duplicated code that can be moved?

Reply:
Actually you are right I just deleted the 102,103,  109,110 and the function has same features of them